logic.py

```diff
@@ -0,0 +1,207 @@
+            request.save()
+        except Exception as e:
+            # save error message in db
+            request.response = str(e)
+            request.status = PreprintMerrittRequests.SubmissionStatus.SEND_ERROR
+            request.save()
+            raise
+        print("DONE")
+
+"""
+Create a zip file for upload to Merritt by combining metadata from OAI endpoint and article file
+"""
+class ZipForPreprint:
+    preprint = None
+    repo = None
+    tmpfolder = None
+    metadata_urlbase = '"https://{param1}/api/oai/?verb=GetRecord&identifier=oai:{param2}:id:{param3}&metadataprefix=oai_dc"'
+    filepath_base = "/apps/eschol/janeway/src/files/"
+
+    def __init__(self, ppobj, repoobj):
+        print("creating zip for preprint")
+        self.preprint = ppobj
+        self.repo = repoobj
+
+    def createZip(self):
+        print("creating zip file")
+        # create temp folder if needed
+        self.tmpfolder = django_settings.MERRITT_TMP + str(self.preprint.id)
+        zipfile = f'{django_settings.MERRITT_TMP}{self.preprint.id}.zip'
+        print(self.tmpfolder)
+        # create folder if needed
+        os.system(f'mkdir -p {self.tmpfolder}')
+        # empty the folder if needed
+        os.system(f'rm {self.tmpfolder}/*')
+
+        # copy files temp folder
+        self.copyArticle()
+        self.generateMetadata()
+
+        # generate zip file  
+        zipcommand = f'zip -r {zipfile} {self.tmpfolder}/'
+        print(zipcommand)
+        result = os.system(zipcommand)
+        assert(result == 0)
+        return zipfile
+
+    def copyArticle(self):
+        print("copy article file")
+        print(self.preprint.title)
+        print(self.preprint.submission_file.file)
+        fullpath = self.filepath_base + str(self.preprint.submission_file.file)
+        temppath = self.tmpfolder + '/' + Path(fullpath).name
+        print("copy from " + fullpath)
+        print("copy to " + temppath)
+        result = os.system("cp {} {}".format(fullpath, temppath))
+        assert(result == 0)
+
+    def generateMetadata(self):
+        print("generate meta data")
+        print(self.repo.short_name)
+
+        # get url from repo
+        metadata_url = self.metadata_urlbase.format(param1=self.repo.domain, param2= self.repo.short_name, param3=self.preprint.id)
+        print(metadata_url)
+
+        # save the metadata in temp folder
+        xmlname = f'{self.tmpfolder}/meta_{self.preprint.id}.xml'
+        result = os.system("curl {} > {}".format(metadata_url, xmlname))
+        assert(result == 0)
+    
+
+"""
+Send zip to Merritt
+"""
+class MerrittForPreprint:
+    # get merritt user name, password and url from settings
+    merritt_base = 'curl -u {param1} -F "file=@{param2}" -F "type=container" -F "submitter={param3}" "title={param4}" --form-string "creator={param5}" -F "responseForm=xml" -F "profile={param6}" -F "localIdentifier={param7}" "{param8}"'
```

Comment:
I would recommend using urllib (like in the EZID plugin) for sending this request.  I think error handling and checking would be better. At a minimum I'd recommend giving the parameters more descriptive names.

Reply:
I agree, using Python methods where we can would be preferable (and more resilient) than a system call.

Reply:
I'll update the param name for now and add a trello card for moving away from curl.

Reply:
https://trello.com/c/aEqJR6lr/1184-preprint-merritt-plugin-enhancements-pubd-1184